### PR TITLE
Please review loadData() and splitTrainValid functions

### LINKED ISSUE #20: Data Enhacements
- [ ] define method for reading data directly from Kaggle api.
- [ ] add validation subset to data processing method.
- [ ] modify training pipeline as required.

### PR DESCRIPTION
**Pull request by nimeifu**
Wrote a function to migrate read_all() into class Bragi, called it loadData(), also wrote a function that splits processed data into 80% training set and 20% validation set.

### CODE REVIEW

File: src/__init__.py [hunk 1]
```diff
@@ -16,163 +19,244 @@
...
+
+        Args:
+            data (dict): Outputs from processData()
+
+        Returns:
+            dict: Dataframe with Input & Label, unique label series,
+            training set with 80% of data and validation set with 20% of data
+
+        Example:
+            >>> agent = Bragi(config=config)
```
Comment (GabeSantini) [inline src/__init__.py]: I mentioned the requirement was not to paste the class into chatgpt...

File: src/__init__.py [hunk 2]
```diff
@@ -16,163 +19,244 @@
...
+        # vocabulary size
+        for text in df['lyrics']:
+            words = text.split()
+            for word in words:
+                self.unique_words.add(word)
+        self.vocab_size: int = len(self.unique_words)
+
+        return {"population": df[['lyrics', 'mood']], "labels": self.labels}
+
+    def splitTrainValid(self, data: dict) -> dict:
```
Comment (GabeSantini) [inline src/__init__.py]: Rather than having to call this separately, split the data within the process method and add dict items to store them separately. I would suggest adding attributes to store them as well. Remember validation and testing are 2 separate things. Modify accordingly.

File: src/__init__.py [hunk 3]
```diff
@@ -16,163 +19,244 @@
...
+            "labels": data['labels'],
+
+            "splits": {
+                "train": train_df,
+                "validation": val_df
             }
-            ...
+        }
+
+    def train(self, data,) -> dict:
```
Comment (GabeSantini) [inline src/__init__.py:144]: If you are splitting the data in another method, include the steps leading up to it in the data processing as well.

File: src/__init__.py [hunk 4]
```diff
@@ -16,163 +19,244 @@
...
+            # TODO: Discuss looping thorugh with various dropout rates
+            tf.keras.layers.Dropout(rate=self.dropout),
+            # TODO: replace activation values with parameter
+            tf.keras.layers.Dense(units=len(self.labels),
+                                  activation=self.activation)
+        ])
+
+        # Step 5: Compile and train
+        self.model.compile(
+            loss=self.loss, optimizer=self.optimizer, metrics=self.metrics)
```
Comment (GabeSantini) [inline src/__init__.py]: Add validation parameter and point it to validation data

**Review by GabeSantini — CHANGES_REQUESTED** [review]
Gj but not quite, needs refinement. You should take a moment to read through the class and identify how other methods are configured. Your change would introduce redundancy.
